roofit/roofit/src/RooGaussExpTails.cxx

```diff
@@ -0,0 +1,110 @@
+/** \class RooGaussExpTails
+    \ingroup Roofit
+
+PDF implementing a Gaussian core + double-sided exponential tail distribution
+\author Souvik Das (8/1/2013) Initial implementation and
+Giovanni Marchiori (30/3/2016) Implemented analytic integral
+\see http://arxiv.org/pdf/1603.08591v1.pdf, https://github.com/mpuccio/RooCustomPdfs/blob/master/RooGausDExp.cxx, https://doi.org/10.1142/S0217751X14300440
+\note To use the one-sided version, just set the opposite parameter k to a very large value
+*/
+
+
+#include "RooGaussExpTails.h"
+#include "RooAbsReal.h"
+#include <cmath>
+#include "Math/ProbFuncMathCore.h"
+
+ClassImp(RooGaussExpTails)
+
+
+//_____________________________________________________________________________
+RooGaussExpTails::RooGaussExpTails(const char *name, const char *title,
+                            RooAbsReal::Ref _x,
+                            RooAbsReal::Ref _x0,
+                            RooAbsReal::Ref _sigma,
+                            RooAbsReal::Ref _kL,
+                            RooAbsReal::Ref _kH) :
+     RooAbsPdf(name,title),
+     x_("x","x",this,_x),
+     x0_("x0","x0",this,_x0),
+     sigma_("sigma","sigma",this,_sigma),
+     kL_("kL","kL",this,_kL),
+     kH_("kH","kH",this,_kH)
+{
+}
+
+
+//_____________________________________________________________________________
+RooGaussExpTails::RooGaussExpTails(const RooGaussExpTails& other, const char* name) :
+     RooAbsPdf(other,name),
+     x_("x",this,other.x_),
+     x0_("x0",this,other.x0_),
+     sigma_("sigma",this,other.sigma_),
+     kL_("kL",this,other.kL_),
+     kH_("kH",this,other.kH_)
+{
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+namespace {
+
+inline double gaussianIntegral(double tmin, double tmax)
+{
+   double m_sqrt_2_pi = 2.50662827463;//std::sqrt(TMath::TwoPi())
+   return m_sqrt_2_pi*(ROOT::Math::gaussian_cdf(tmax) - ROOT::Math::gaussian_cdf(tmin));
+}
+
+inline double tailIntegral(double tmin, double tmax, double k)
+{
+   double a = std::exp(0.5*k*k)/k;
+   return (a*(std::exp(k*tmax)-std::exp(k*tmin)));
+}
+
+} // namespace
+
+
+
+//_____________________________________________________________________________
+Double_t RooGaussExpTails::evaluate() const
+{
+   Double_t t=(x_-x0_)/sigma_;
+
+   if (t<=-kL_)
+      return exp(kL_*kL_/2.+kL_*t);
```

Comment:
Minor detail waiting for the expert eye of @guitargeek : would it make sense to multiply by 0.5 instead of dividing?